Source/ARTPushActivationState.m

```diff
@@ -117,6 +118,12 @@ - (ARTPushActivationState *)transition:(ARTPushActivationEvent *)event {
     }
     else if ([event isKindOfClass:[ARTPushActivationEventCalledActivate class]]) {
         [self.machine registerForAPNS];
+#if TARGET_OS_IOS
+        ARTLocalDevice *const local = self.machine.rest.device_nosync;
+        if ([local clientId] == nil) {
```

Comment:
Why the nil check? RSH8b doesn't describe this.

Reply:
https://github.com/ably/ably-cocoa/pull/1842/commits/ed9f576e68e0c70426a6db7af78b5c0e9cf63f69

Reply:
>If so, should we not remove it?

You are right, removing `clientId` is optional as RSH8b is not implemented. Reverting https://github.com/ably/ably-cocoa/pull/1842/commits/7dfc6c6351821d3e230bafa92da1ec0a51f5710a